### PR TITLE
Implement AuthMeBungee autologin

### PR DESCRIPTION
**Pull request by sgdc3**
There is a failing test due to cyclic dependency injection, @ljacqu could you take a quick look at this?

### CODE REVIEW

**Comment by ljacqu** [comment]
hey @sgdc3, you have a cyclic dependency. Your only choice is to split the BungeeService. The problem is that now you inject Management into BungeeService, but BungeeService gets injected into stuff that Management requires (e.g. AsyncQuit).

So in other words now the injector is like "to create BungeeService I need Management" -> "to create management I need asyncquit" -> "to create asyncquit I need BungeeService—uh oh"

File: src/main/java/fr/xephi/authme/service/bungeecord/BungeeService.java [hunk 1]
```diff
@@ -113,9 +117,21 @@ public void onPluginMessageReceived(String channel, Player player, byte[] data)
             case MessageType.REFRESH:
                 dataSource.refreshCache(name);
                 break;
+            case MessageType.BUNGEE_LOGIN:
+                handleBungeeLogin(name);
```
Comment (ljacqu) [inline src/main/java/fr/xephi/authme/service/bungeecord/BungeeService.java]: missing a `break` here

**Comment by HexelDev** [comment]
@ljacu that example is awesome ahhahaha

**Comment by sgdc3** [comment]
>Your only choice is to split the BungeeService.

Any suggestion?

**Comment by ljacqu** [comment]
> Any suggestion?

Sending and receiving. Inside the Management dependencies you're only sending out messages. The only annoying thing about that is the channel handling, which I guess you could keep in BungeeService maybe.
Without looking at it into too much detail I reckon you could do something like the following.

**Comment by ljacqu** [comment]
img_7426

File: src/main/java/fr/xephi/authme/service/bungeecord/BungeeService.java [hunk 2]
```diff
@@ -56,10 +48,14 @@ public void reload(Settings settings) {
             messenger.registerOutgoingPluginChannel(plugin, "BungeeCord");
         }
         if (!messenger.isIncomingChannelRegistered(plugin, "BungeeCord")) {
-            messenger.registerIncomingPluginChannel(plugin, "BungeeCord", this);
+            messenger.registerIncomingPluginChannel(plugin, "BungeeCord", injector.getSingleton(BungeeReceiver.class));
```
Comment (sgdc3) [inline src/main/java/fr/xephi/authme/service/bungeecord/BungeeService.java]: Is that correct? @ljacqu  :/

**Comment by ljacqu** [comment]
I see the issue and don't have a solution :/ I need to check this out this evening on a PC

**Comment by ljacqu** [comment]
I pushed my proposal. I haven't tested it in-game. Please make any necessary adjustments and kindly _squash_ this PR if you decide to merge it. We don't need six commits of back and forth in our Git history :)

**Comment by sgdc3** [comment]
Looks good @ljacqu

**Pull request merged**